.github/workflows/e2e-tests-ci.yml

```diff
@@ -57,3 +53,23 @@ jobs:
       - name: ci/playwright/npm-check
         run: |
           npm run check
+  smoketests:
+    runs-on: ubuntu-22.04
+    needs:
+      - cypress-check
+      - playwright-check
+    defaults:
+      run:
+        working-directory: e2e-tests
+    steps:
+      - name: ci/checkout-repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
Since this is going to be triggered by Argo Events we should change the logic a bit . We need to integrate our new custom actions . 
Check [the actions here](https://github.com/mattermost/actions/pull/9) and [this](https://github.com/mattermost/delivery-platform/pull/1/files#diff-8d362d508e41530f3b7f9fe2d238f218bef5c4f5b51b339bfadc11d44722ccdc) workflow file. 

We need to add these steps below
```yaml
  expose-incoming-variables:
    runs-on: ubuntu-22.04
    outputs:
      repository_full_name: ${{ env.TRIGGERER_REPO_FULL_NAME }}
      commit_sha: ${{ env.TRIGGERER_COMMIT_SHA }}
      tag: ${{ env.TRIGGERER_BRANCH }}
    steps:
      - uses: mattermost/actions/release/expose-events-vars@f04d8cc6fbd215659c3406974a74f7167c00803a
        with:
          payload: ${{ inputs.payload }}
```

We need to test if we want to manually update the status check with the below step . I think we will not have to since if the check is running in the same commit sha it will show up in the PR. 
```yaml
  update-initial-status:
    runs-on: ubuntu-22.04
    needs:
      - expose-incoming-variables
    steps:
      - uses: mattermost/actions/release/update-commit-status@f04d8cc6fbd215659c3406974a74f7167c00803a
        with:
          repository_full_name: ${{ needs.expose-incoming-variables.outputs.repository_full_name }}
          commit_sha: ${{ needs.expose-incoming-variables.outputs.commit_sha }}
          context: e2e/tests
          description: E2E tests for mattermost-server
          status: pending
```

In any case we need to change the checkout actions with 
```yaml
      - name: ci/checkout-repo
        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
        with:
          repository: ${{ needs.expose-incoming-variables.outputs.repository_full_name }}
          ref: ${{ needs.expose-incoming-variables.outputs.commit_sha }}
```

Reply:
Thank you Antonis, I applied your suggested changes!

Pairing up sounds great, tomorrow it's a public holiday in Germany but I would gladly be in for Tuesday

Reply:
Example repo [here](https://github.com/mattermost/test-public-project/pull/6)